### PR TITLE
Add static initialize method to fix plugin tests

### LINKED ISSUE #9535: private method `initialize' called for LogStash::Logging::Logger:Class
All plugin tests are failing currently for logstash 6.x/master as [logstash-devutils's rspec_helper calls](https://github.com/elastic/logstash-devutils/blob/master/lib/logstash/devutils/rspec/spec_helper.rb#L27):

`LogStash::Logging::Logger::initialize("file:///" + log4j2_properties)`

But it seems not be available anymore, causing the tests to fail. [Here is an example](https://travis-ci.org/logstash-plugins/logstash-filter-date/jobs/375058466#L1164-L1172):

```
Failure/Error: # encoding: utf-8
NoMethodError:
  private method `initialize' called for LogStash::Logging::Logger:Class
  Did you mean?  initialize_dup
# /home/travis/.rvm/gems/jruby-9.1.13.0/gems/logstash-devutils-1.3.6-java/lib/logstash/devutils/rspec/spec_helper.rb:27:in `<main>'
# ./spec/filters/date_spec.rb:1:in `<main>'
# ./spec/filters/date_spec.rb:3:in `(root)'
No examples found.
```

### PR DESCRIPTION
**Pull request by danhermann**
Adds static `intialize` method and `info?` instance method.

Fixes #9535.

### CODE REVIEW

**Comment by original-brownbear** [comment]
@danhermann LGTM if it goes green :)

File: logstash-core/src/main/java/org/logstash/log/LoggerExt.java [hunk 1]
```diff
@@ -31,10 +31,8 @@ public LoggerExt(final Ruby runtime, final RubyClass metaClass) {
     }
 
     @JRubyMethod(meta = true)
-    public static LoggerExt initialize(final ThreadContext context, final IRubyObject self, final IRubyObject loggerName) {
-        LoggerExt logger = new LoggerExt(context.runtime, (RubyClass) self);
-        logger.logger = LogManager.getLogger(loggerName.asJavaString());
-        return logger;
+    public static IRubyObject initialize(final ThreadContext context, final IRubyObject self, final IRubyObject configPath) {
```
Comment (original-brownbear) [inline logstash-core/src/main/java/org/logstash/log/LoggerExt.java]: @danhermann you can just do:

```
 @JRubyMethod(meta = true, name = {"reconfigure", "initialize"})
```

on the annotation for `reconfigure` instead to define the alias. No need to repeat the signature `name` takes an array annotation argument :)

**Comment by original-brownbear** [comment]
@danhermann still LGTM

**Comment by yaauie** [comment]
Fix LGTM, and build is green :tada:

**Comment by danhermann** [comment]
Thanks, @yaauie and @original-brownbear!

**Pull request closed without merging**